From 3d14b404f48daff4fc9b11f5dd775b325e009e41 Mon Sep 17 00:00:00 2001
From: firo yang <firogm@gmail.com>
Date: Mon, 1 Jul 2013 21:01:07 +0800
Subject: [PATCH] Drivers md: fix a potential NULL pointer dereference

In function do_region(), when  bio = bio_alloc_bioset() return NULL will
lead a potential NULL pointer dereference.

Signed-off-by: Firo Yang <fiorgm@gmail.com>
---
 drivers/md/dm-io.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/md/dm-io.c b/drivers/md/dm-io.c
index ea49834..c9bc6ca 100644
--- a/drivers/md/dm-io.c
+++ b/drivers/md/dm-io.c
@@ -305,6 +305,8 @@ static void do_region(int rw, unsigned region, struct dm_io_region *where,
 					  dm_sector_div_up(remaining, (PAGE_SIZE >> SECTOR_SHIFT)));
 
 		bio = bio_alloc_bioset(GFP_NOIO, num_bvecs, io->client->bios);
+		if (unlikely(!bio))
+			return;
 		bio->bi_sector = where->sector + (where->count - remaining);
 		bio->bi_bdev = where->bdev;
 		bio->bi_end_io = endio;
-- 
1.8.3.1

